src/shogun/features/LatentFeatures.cpp

```diff
@@ -78,16 +78,4 @@ void CLatentFeatures::init()
 {
 	SG_ADD((CSGObject**) &m_samples, "samples", "Array of examples",
 			MS_NOT_AVAILABLE);
-}
-
-CLatentFeatures* CLatentFeatures::obtain_from_generic(CFeatures* base_feats)
```

Comment:
as you can see, the current `obtain_from_generic` methods are inconsistent: some of the increase the reference counter, and some of them dont. This one doesnt.

`as` does NOT increase the reference counter. This explains all the segfaults in the CI.
Unfortunately, if you replace a `obtain_from_generic` that increased the reference counter, you will have to add a `SG_REF` after the call.
However, I actually think that neither `as` nor `obtain_from_generic` should increase the reference counter at all, so the best would be just to modifz the code downstream to not decrease the reference counter downstream